code/obj/machinery/computer/general_air_control.dm

```diff
@@ -503,62 +493,27 @@ Rate: <A href='?src=\ref[src];change_vol=-10'>--</A> <A href='?src=\ref[src];cha
 		//boutput(world, "[id] received a signal from [id_tag]!")
 		mixer_information = signal.data
 
-	proc/return_text()
-		var/mixer_data
-		if(mixerid)
-			var/long_name = mixerid
-			var/list/data = mixer_information
-			var/mixer_part = ""
-
-			if(data)
-				mixer_part += "<B>Input 1 Composition</B>: <BR>"
-				GET_SIGNAL_MIXTURE("In1")
-				if(data["in1tg"])
-					mixer_part += "<FONT color='black'>[data["in1tg"]]% OTHER</FONT>   "
-				if (data["in1kpa"] && data["in1temp"])
-					mixer_part += "<br>Pressure: [data["in1kpa"]] kPa / Temperature: [data["in1temp"]] &deg;C"
-				mixer_part += "<BR>"
-
-				mixer_part += "<B>Input 2 Composition</B>: <BR>"
-				GET_SIGNAL_MIXTURE("In2")
-				if(data["in2tg"])
-					mixer_part += "<FONT color='black'>[data["in2tg"]]% OTHER</FONT>   "
-				if (data["in2kpa"] && data["in2temp"])
-					mixer_part += "<br>Pressure: [data["in2kpa"]] kPa / Temperature: [data["in2temp"]] &deg;C"
-
-				mixer_part += "<hr>"
-
-				mixer_part += "<b>Output Target Pressure</b>: <A href='?src=\ref[src];pressure_adj=-100'>-</A> <A href='?src=\ref[src];pressure_adj=-10'>-</A> <A href='?src=\ref[src];pressure_set=1'>[data["target_pressure"]] kPa</A> <A href='?src=\ref[src];pressure_adj=10'>+</A> <A href='?src=\ref[src];pressure_adj=100'>+</A><BR>"
-				mixer_part += "<b>Pump Status</b>: <A href='?src=\ref[src];toggle_pump=1'>[data["pump_status"]]</A><BR>"
-				mixer_part += "<B>Gas Input Ratio</b>: <A href='?src=\ref[src];ratio=5'><<</A> <A href='?src=\ref[src];ratio=1'><</A> [data["i1trans"]]% /  [data["i2trans"]]% <A href='?src=\ref[src];ratio=-1'>></A> <A href='?src=\ref[src];ratio=-5'>>></A>"
-
-				mixer_part += "<HR><B>Resulting Composition</B>: <BR>"
-				GET_SIGNAL_MIXTURE("Out")
-				if(data["outtg"])
-					mixer_part += "<FONT color='black'>[data["outtg"]]% OTHER</FONT>   "
-				if (data["outkpa"] && data["outtemp"])
-					mixer_part += "<br>Pressure: [data["outkpa"]] kPa / Temperature: [data["outtemp"]] &deg;C"
-				mixer_part += "<BR>"
-
-				mixer_data += mixer_part
+		tgui_process.update_uis(src)
 
-			else
-				mixer_part = "<FONT color='red'>[long_name] can not be found!<A href='?src=\ref[src];refresh_status'>Search</A></FONT><BR>"
+	ui_interact(mob/user, datum/tgui/ui)
+		ui = tgui_process.try_update_ui(user, src, ui)
+		if (!ui)
+			ui = new(user, src, "GasMixer")
+			ui.open()
 
-				mixer_data += mixer_part
+	ui_static_data(mob/user)
+		. = ..()
+		.["name"] = name
+		.["mixerid"] = mixerid
+		.["MAX_PRESSURE"] = MAX_PRESSURE
```

Comment:
While these don't change much, this is such a small amount of data that it seems silly not to send it up on `ui_data` instead, guaranteeing consistency.